src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -0,0 +1,388 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.App;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    // Can pass to symbol table at compile time, and then it has a handle
+    // if we use it for execution at runtime. 
+    public interface ICanGetValue
+    {
+        FormulaValue Value { get; }
+    }
+
+    // NameInfo data to associate a symbol at bind time with a runtime config at runtime. 
+    internal class NameSymbol
```

Comment:
SymbolName?

Reply:
No, this is a symbol that's a name. SymbolName would read as the name of a symbol, at least to me

Reply:
Should it be NamedSymbol then? (with a d)

Reply:
this is consistent with NameLookupInfo